vignettes/articles/scripting-tasks-config.Rmd

```diff
@@ -0,0 +1,387 @@
+---
+title: "Scripting task configuration files"
+---
+
+```{r, setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+## Introduction
+
+The `tasks.json` configuration file is a the source of truth for validating a
+hub and is also one of the most complex aspects of a hub. This configuration
+file is a formal representation of an official collaborative forecasting
+challenge that allows direct and unambiguous evaluation and comparison of model
+outputs. It allows hub administrators to set a clear standard for model
+submissions.
+
+While we provide [a hubTemplate](https://github.com/hubverse-org/hubTemplate/)
+for use to get started with setting up a hub, JSON files are easy for a computer
+to read, but not so easy for a human to read and less so to write. The hard part
+of creating a `tasks.json` file should be what model parameters to use, not
+constantly looking at JSON syntax or hunting down a missing comma.
+To make this process easier, **we provide the `create_*()` family of functions
+for creating `tasks.json` files programmatically**.  These can be useful for
+creating your initial `task.json` file, amending an existing file's
+configurations or even appending new rounds programmatically. By making use of
+this functionality, hubs can even automate the process of updating their files
+through Github Actions.  For example, the [variant nowcast
+hub](https://github.com/reichlab/variant-nowcast-hub/tree/main) uses a [GitHub
+Action
+workflow](https://github.com/reichlab/variant-nowcast-hub/blob/29a9265d0b30f385fc47195d3d7f65ecda7a41d2/.github/workflows/create-modeling-round.yaml#L47)
+that [runs a
+script](https://github.com/reichlab/variant-nowcast-hub/blob/main/src/make_round_config.R)
+to append a new round with updated variants to model on a weekly schedule!
+
+In this vignette, we will use the config creation functions in `hubAdmin` to
+create a `tasks.json` file that represents two rounds of a modeling effort to
+predict influenza hospitalizations and optionally deaths 1 to 4 weeks ahead
+with mean, median, and quantile predictions in the US, and optionally 5 states.
+
+Before getting to the scripting, it's important to get a general look at what
+a `tasks.json` file looks like structurally.
+
+## Structure of a `tasks.json` configuration file
+
+The structure of a `tasks.json` configuration file roughly looks like the
+diagram below:
+
+```
+tasks.json
+├─schema_version: "https://.../v4.0.0/tasks-schema.json"
+├─rounds:
+│ ├─round_id_from_variable: true
+│ ├─round_id: origin_date
+│ ├─model_tasks:
+│ │   ├─task_ids:
+│ │   │ └─[...]
+│ │   ├─output_type:
+│ │   │ └─[...]
+│ │   └─target_metadata:
+│ │     └─[...]
```

Comment:
The different amount of indentation for the branch under `model_tasks` (relative to the branches under `tasks.json`, `rounds`, etc.) stuck out to me.  Is there a particular reason for that choice?

Reply:
sorry no I fucked up

Reply:
fixed in 01484bd6a15457ac3af49b5c5e600b138d4cbc98